xray/sqlcontext_test.go

```diff
@@ -0,0 +1,354 @@
+	assert.Equal(t, "", subseg.SQL.URL)
+	assert.Equal(t, dsn, subseg.SQL.ConnectionString)
+	assert.Equal(t, "test version", subseg.SQL.DatabaseVersion)
+	assert.Equal(t, "test user", subseg.SQL.User)
+	assert.False(t, subseg.Throttle)
+	assert.False(t, subseg.Error)
+	assert.False(t, subseg.Fault)
+}
+
+func TestOracle(t *testing.T) {
+	dsn := "test-oracle"
+	db, mock, err := sqlmock.NewWithDSN(dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+	mockPostgreSQL(mock, errors.New("syntax error"))
+	mockMySQL(mock, errors.New("syntax error"))
+	mockMSSQL(mock, errors.New("syntax error"))
+	mockOracle(mock, nil)
+
+	subseg, err := capturePing(dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.NoError(t, mock.ExpectationsWereMet())
+
+	assert.Equal(t, "remote", subseg.Namespace)
+	assert.Equal(t, "Oracle", subseg.SQL.DatabaseType)
+	assert.Equal(t, "", subseg.SQL.URL)
+	assert.Equal(t, dsn, subseg.SQL.ConnectionString)
+	assert.Equal(t, "test version", subseg.SQL.DatabaseVersion)
+	assert.Equal(t, "test user", subseg.SQL.User)
+	assert.False(t, subseg.Throttle)
+	assert.False(t, subseg.Error)
+	assert.False(t, subseg.Fault)
+}
+
+func TestUnknownDatabase(t *testing.T) {
+	dsn := "test-unknown"
+	db, mock, err := sqlmock.NewWithDSN(dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+	mockPostgreSQL(mock, errors.New("syntax error"))
+	mockMySQL(mock, errors.New("syntax error"))
+	mockMSSQL(mock, errors.New("syntax error"))
+	mockOracle(mock, errors.New("syntax error"))
+
+	subseg, err := capturePing(dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.NoError(t, mock.ExpectationsWereMet())
+
+	assert.Equal(t, "remote", subseg.Namespace)
+	assert.Equal(t, "Unknown", subseg.SQL.DatabaseType)
+	assert.Equal(t, "", subseg.SQL.URL)
+	assert.Equal(t, dsn, subseg.SQL.ConnectionString)
+	assert.Equal(t, "Unknown", subseg.SQL.DatabaseVersion)
+	assert.Equal(t, "Unknown", subseg.SQL.User)
+	assert.False(t, subseg.Throttle)
+	assert.False(t, subseg.Error)
+	assert.False(t, subseg.Fault)
+}
+
+func TestStripPasswords(t *testing.T) {
+	tc := []struct {
+		in   string
+		want string
+	}{
+		{
+			in:   "user=user database=database",
+			want: "user=user database=database",
+		},
+		{
+			in:   "user=user password=password database=database",
+			want: "user=user database=database",
+		},
+		{
+			in:   "odbc:server=localhost;user id=sa;password={foo}};bar};otherthing=thing",
+			want: "odbc:server=localhost;user id=sa;otherthing=thing",
+		},
+
+		// see https://github.com/aws/aws-xray-sdk-go/issues/181
+		{
+			in:   "password=",
+			want: "",
+		},
+		{
+			in:   "pwd=",
+			want: "",
+		},
+
+		// test cases for https://github.com/go-sql-driver/mysql
+		{
+			in:   "user:password@tcp(localhost:5555)/dbname?tls=skip-verify&autocommit=true",
+			want: "user@tcp(localhost:5555)/dbname?tls=skip-verify&autocommit=true",
+		},
+
+		// got: "@tcp(localhost:5555)/dbname?tls=skip-verify&autocommit=true"
```

Comment:
you can uncomment this test cases after adding the above suggested logic.

Reply:
Also, I have merged this PR (https://github.com/aws/aws-xray-sdk-go/pull/191) to immediately unblock the customer. After we add this we can overwrite those changes.